### ISSUE TITLE
@Builder does not generate a class compliant with the JavaBeans Specification 

### ISSUE THREAD

**Issue by helara** [rzwitserloot/lombok#1057]
Please make (at least in an optional way) the generated Builder class compliant with the JavaBeans Specification.

Currently, the generated setter for a field "T foo", is foo(T value). This makes it impossible to use in a automatic mapping (SpringMVC RequestBody, Hibernate, for instance). 

Thanks.

**Comment by Maaartinus** [comment]
IIRC Javabeans forces setters to return `void`, so you won't get any usable builder using them. So you'd better need two different setters, so the tools are happy and you can still write things like `C.builder().x(1).y(2).build()`. This most probably should not be the default behavior. For your case I'd suggest a new option like

    @Builder(javabean=true)

**Comment by helara** [comment]
@Maaartinus Yes, that is exactly what I was thinking about. An option like that would be extremely useful for constructing inmutable objects/entities from external sources. Ex: 

1. `interface MySpringDataRepository extends JpaRepository<MyInmutableEntity.Builder, String>`

2. `@RequestMapping("/resource")
  ResourceDto create(@RequestBody @Valid Resource_CreateRequest.Builder request)`

Please note how `MyInmutableEntity` and `Resource_CreateRequest` are inmutable once they are built. IMHO, this is a very useful use case for a `@Builder(javabean=true)` option.
